plugins/ote-migration/commands/migrate.md

```diff
@@ -0,0 +1,2471 @@
+# Add OTE dependency if not already present
+if ! grep -q "github.com/openshift-eng/openshift-tests-extension" go.mod; then
+    go get github.com/openshift-eng/openshift-tests-extension@latest
+fi
+
+echo "Step 2: Add replace directive for test/e2e package..."
+# Add replace directive to make test/e2e resolve correctly
+if ! grep -q "replace.*$MODULE_NAME/test/e2e" go.mod; then
+    # Check if replace section exists
+    if grep -q "^replace (" go.mod; then
+        # Add to existing replace section (after "replace (" line)
+        sed -i "/^replace (/a\\	$MODULE_NAME/test/e2e => ./test/e2e" go.mod
+    else
+        # Create new replace section
+        echo "" >> go.mod
+        echo "replace $MODULE_NAME/test/e2e => ./test/e2e" >> go.mod
+    fi
+    echo "✅ Replace directive added"
+fi
+
+echo "Step 3: Vendor dependencies..."
+# Vendor all dependencies (including from test/e2e package)
+go mod vendor
+
+echo "✅ Root go.mod updated with OTE dependency and test/e2e replace directive"
+```
+
+**Note:** The monorepo strategy is much simpler:
+- test/e2e and test/testdata are regular packages in the main module
+- NO separate test/e2e/go.mod file
+- Just add OTE dependency to root go.mod
+- Add replace directive for test/e2e => ./test/e2e
+- Vendor all dependencies
+
+**For Single-Module Strategy:**
+
+Create `tests-extension/go.mod` following proper Go module initialization sequence:
+```bash
+cd <working-dir>/tests-extension
+
+echo "Step 1: Initialize Go module..."
+# Step 1: go mod init - Creates go.mod with module declaration only
+go mod init github.com/<org>/<extension-name>-tests-extension
+
+echo "Step 2: Add required dependencies..."
+# Step 2: Add dependencies (go get will update go.mod and create go.sum)
+# Get the correct openshift/origin version from openshift-tests-private
+ORIGIN_VERSION=$(grep "github.com/openshift/origin" "$OTP_PATH/go.mod" | head -1 | awk '{print $2}')
+echo "Using openshift/origin version: $ORIGIN_VERSION (from openshift-tests-private)"
+
+# Add dependencies with retry logic for network issues
+echo "Adding openshift-tests-extension dependency..."
+if ! go get github.com/openshift-eng/openshift-tests-extension@latest; then
+    echo "⚠️  Warning: Failed to download openshift-tests-extension, retrying..."
+    sleep 2
+    go get github.com/openshift-eng/openshift-tests-extension@latest || echo "❌ Failed after retry"
+fi
+
+echo "Adding openshift/origin dependency..."
+if ! go get "github.com/openshift/origin@$ORIGIN_VERSION"; then
+    echo "⚠️  Warning: Failed to download openshift/origin, retrying..."
+    sleep 2
+    go get "github.com/openshift/origin@$ORIGIN_VERSION" || echo "❌ Failed after retry"
+fi
+
+echo "Adding Ginkgo and Gomega dependencies..."
+if ! go get github.com/onsi/ginkgo/v2@latest; then
+    echo "⚠️  Warning: Failed to download ginkgo, retrying..."
+    sleep 2
+    go get github.com/onsi/ginkgo/v2@latest || echo "❌ Failed after retry"
+fi
```

Comment:
This needs to be the openshift fork

Reply:
Add a separate step to extract and add replace directives from openshift-tests-private

Reply:
run-test of build binary tested passed in https://github.com/openshift/router/pull/705